core/src/test/java/com/linecorp/armeria/client/HttpClientIntegrationTest.java

```diff
@@ -767,6 +768,14 @@ void responseAbortWithException() throws InterruptedException {
                 .hasCause(badState);
     }
 
+    @Test
+    void httpsRequstToPlainTextEndpoint() throws Exception {
+        final WebClient client = WebClient.builder(SessionProtocol.HTTPS, newEndpoint())
+                                          .factory(ClientFactory.insecure()).build();
+        assertThatThrownBy(() -> client.get("/hello/world").aggregate().get())
```

Comment:
Can we also test for hitting an ssl endpoint without a proper cert? I think making server self-signed and not setting the clientfactory to insecure should work.

Reply:
excellent point. done 👍